OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -188,37 +187,34 @@ public int CalculateRegionValue(CellRegion sourceRegion)
 			foreach (var cell in sourceRegion.CellCoords)
 			{
 				var mcell = cell.ToMPos(Map);
-				if (Map.Resources.Contains(mcell) && Map.Resources[mcell].Type != 0)
-				{
-					resourceValueInRegion++;
-					var rcell = Map.Resources[mcell];
-					if (ResourceTypesByIndex.TryGetValue(rcell.Type, out var resourceType) && resourceValues.TryGetValue(resourceType, out var resourceValuePerUnit))
-						resourceValueInRegion += Tiles[mcell].Density * resourceValuePerUnit;
-				}
+				byte resource;
+				if (Map.Resources.Contains(mcell)
+					&& (resource = Map.Resources[mcell].Type) != 0
+					&& ResourceTypesByIndex.TryGetValue(resource, out var resourceType)
+					&& resourceValues.TryGetValue(resourceType, out var resourceValuePerUnit))
+					resourceValueInRegion += Tiles[mcell].Density * resourceValuePerUnit;
 			}
 
 			return resourceValueInRegion;
 		}
 
-		protected virtual int CalculateCellDensity(ResourceLayerContents contents, CPos c)
+		protected virtual int CalculateCellDensity(ResourceLayerContents contents, CPos cell)
 		{
 			var resources = Map.Resources;
-			if (contents.Type == null || !info.ResourceTypes.TryGetValue(contents.Type, out var resourceInfo) || resources[c].Type != resourceInfo.ResourceIndex)
+			if (contents.Type == null || !info.ResourceTypes.TryGetValue(contents.Type, out var resourceInfo) || resources[cell].Type != resourceInfo.ResourceIndex)
 				return 0;
 
 			if (!info.RecalculateResourceDensity)
 				return contents.Density.Clamp(1, resourceInfo.MaxDensity);
 
 			// Set density based on the number of neighboring resources
```

Comment:
Can we include a comment like `// This matches the calculation in ResourceLayer that is used ingame`?

Reply:
Added a summary